tests/rust/safe_abstraction/arc_u32.rs

```diff
@@ -0,0 +1,417 @@
+use std::process::abort;
+use std::ptr::NonNull;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+pub struct ArcU32 {
+    ptr: NonNull<ArcInnerU32>,
```

Comment:
My intuition is that starting from the elements closer to the interface of the library and going down, provides people with a smoother learning path. Normally one would look at the interface and/or its documentation and then go deep into the code, so starting from a point they have already some familiarity with would be helpful.

Reply:
Well, readers wanting to first look just at the public API have to (and can easily) ignore the private items/aspects anyway, so that does not seem to be a strong reason for where private items should go.